alf/trainers/evaluator.py

```diff
@@ -410,18 +424,37 @@ def evaluate(env: AlfEnvironment,
         alf.metrics.AverageRewardMetric(
             example_time_step=time_step, buffer_size=num_episodes),
     ]
+
+    counter = 0
+    episode_mode = True
+    if num_episodes > 0:
+        # episode eval mode
+        episodes_per_env = (num_episodes + batch_size - 1) // batch_size
+        env_episodes = torch.zeros(batch_size, dtype=torch.int32)
+        total_num = num_episodes
+    else:
+        episode_mode = False
+        num_eval_steps = num_steps()
+        assert num_eval_steps > 0
+        steps_per_env = (num_eval_steps + batch_size - 1) // batch_size
+        total_num = steps_per_env
+        # additional steps required after ``steps_per_env`` parallel steps over ``batch_size`` envs
+        additional_steps = num_eval_steps - steps_per_env * batch_size
```

Comment:
It seems that this can never be positive? And I feel there is no need to handle this. The extra complexity does not really gain anything meaningful.

Reply:
> It seems that this can never be positive? And I feel there is no need to handle this. The extra complexity does not really gain anything meaningful.

Agree. Simplified the logic in the new commit.